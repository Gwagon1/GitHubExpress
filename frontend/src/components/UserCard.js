// src/components/UserCard.js
import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';

const UserCard = ({ user }) => {
  const [error, setError] = useState(null);

  useEffect(() => {
    if (!user) {
      setError('Failed to fetch user data');
    }
  }, [user]);

  if (error) {
    return <div>{error}</div>;
  }

  if (!user) {
    return <div>Loading...</div>;
  }

  return (
    <div className="user-card">
      <img 
        src={user.avatar_url || 'default-avatar.png'} 
        alt={user.login} 
        className="user-avatar" 
      />
      <div className="user-info">
        <h3>{user.login}</h3>
        <p>{user.name || 'No name available'}</p>
        <Link to={`/user/${user.login}`} className="view-profile">
          View Profile
        </Link>
      </div>
    </div>
  );
};

export default UserCard;
