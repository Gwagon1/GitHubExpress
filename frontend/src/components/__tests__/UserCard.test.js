// src/components/__tests__/UserCard.test.js
import { render, screen, waitFor } from '@testing-library/react';
import UserCard from '../UserCard';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';

describe('UserCard Component', () => {
  const user = {
    login: 'johndoe',
    avatar_url: 'http://example.com/avatar.jpg',
    name: 'John Doe'
  };

  test('renders user card with user data', async () => {
    render(
      <Router>
        <UserCard user={user} />
      </Router>
    );

    // Check if the "Loading..." text is not present (since data is available)
    expect(screen.queryByText('Loading...')).toBeNull();

    // Wait for the user data to be rendered
    await waitFor(() => {
      expect(screen.getByText('John Doe')).toBeInTheDocument();
      expect(screen.getByText('johndoe')).toBeInTheDocument(); // Check for login
      expect(screen.getByRole('img')).toHaveAttribute('src', user.avatar_url); // Check for image
    });
  });

  
  test('displays error message when fetching user data fails', async () => {
    // Simulate passing a missing user prop or an error scenario
    render(<UserCard user={null} />);
  
    // Wait for the error message to appear
    await waitFor(() => {
      expect(screen.getByText('Failed to fetch user data')).toBeInTheDocument();
    });
  });


  test('matches snapshot', async () => {
    const { asFragment } = render(
      <Router>
        <UserCard user={user} />
      </Router>
    );

    // Wait for the user data to be rendered before taking the snapshot
    await waitFor(() => {
      expect(screen.getByText('John Doe')).toBeInTheDocument();
    });

    // Take snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
