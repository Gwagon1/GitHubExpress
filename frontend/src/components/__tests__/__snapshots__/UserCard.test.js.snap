// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="user-card"
  >
    <img
      alt="johndoe"
      class="user-avatar"
      src="http://example.com/avatar.jpg"
    />
    <div
      class="user-info"
    >
      <h3>
        johndoe
      </h3>
      <p>
        John Doe
      </p>
      <a
        class="view-profile"
        href="/user/johndoe"
      >
        View Profile
      </a>
    </div>
  </div>
</DocumentFragment>
`;
